### PR TITLE
Add nfts manual prices to the API latest manual prices response

### PR DESCRIPTION
**Pull request by yabirgb**
Return prices of NFTs stored in the database

### CODE REVIEW

File: rotkehlchen/api/rest.py [hunk 1]
```diff
@@ -4106,6 +4106,26 @@ def get_manual_latest_prices(
                 'to_asset': price_entry[1],
                 'price': price_entry[2],
             })
+
+        if self.rotkehlchen.chains_aggregator.get_module('nfts') is not None:
+            # query also nfts manual prices
+            nft_price_data = self._eth_module_query(
```
Comment (LefterisJP) [inline rotkehlchen/api/rest.py:4112]: Does this also query the nft prices? Or just read from the DB? From a quick look it just reads from the DB, which should be okay.

Also why go through `_eth_module_query()` and not query `get_nfts_with_price` directly? What's the reasoning of the extra work? You are actually reading just the `result` of the dict and not using the constructured api response at all.

File: rotkehlchen/chain/ethereum/modules/nft/nfts.py [hunk 2]
```diff
@@ -280,36 +284,48 @@ def get_nfts_with_price(
             query_str += ' AND is_lp=?'
             bindings.append(lps_handling == NftLpHandling.ONLY_LPS)
 
+        if only_with_manual_prices is True:
```
Comment (LefterisJP) [inline rotkehlchen/chain/ethereum/modules/nft/nfts.py:287]: the filter is getting complicated here. Can't you build a normal filter like we do in the rest of the code?

File: rotkehlchen/chain/ethereum/modules/nft/nfts.py [hunk 3]
```diff
@@ -280,36 +284,48 @@ def get_nfts_with_price(
...
-                    to_asset = Asset(to_asset_id).check_existence()
+                    target_asset = Asset(to_asset_id).check_existence()
                 except UnknownAsset:
                     log.error(
                         f'Unknown asset {to_asset_id} in custom nft price DB table. Ignoring.',
                     )
                     continue
 
-                if to_asset != A_USD:
+                if target_asset != A_USD and only_with_manual_prices is False:
```
Comment (LefterisJP) [inline rotkehlchen/chain/ethereum/modules/nft/nfts.py:312]: So `if target_asset != A_USD and only_with_manual_prices is True:`

then `usd_price` is not really a usd price? Isn't that broken?

File: rotkehlchen/tests/api/test_current_assets_price.py [hunk 4]
```diff
@@ -527,3 +531,78 @@ def test_prices_cache_invalidation_for_manual_prices(rotkehlchen_api_server):
     # # the max diff is to account for price changes between the requests interval
     assert FVal(avax_result['assets']['AVAX'][0]).is_close(FVal(eth_result['assets']['ETH'][0]), max_diff='10')  # noqa: E501
     assert result['target_asset'] == 'USD'
+
+
+@pytest.mark.parametrize('should_mock_current_price_queries', [True])
+@pytest.mark.parametrize('ethereum_accounts', [['0xc37b40ABdB939635068d3c5f13E7faF686F03B65']])
+@pytest.mark.parametrize('ethereum_modules', [['nfts']])
+def test_get_manual_prices_with_nfts(
+    rotkehlchen_api_server: 'APIServer',
```
Comment (LefterisJP) [inline rotkehlchen/tests/api/test_current_assets_price.py:540]: shouldnt' this need 1 more indentation level?

File: rotkehlchen/api/rest.py (hunk 1 above)
Comment (yabirgb) [reply rotkehlchen/api/rest.py:4112]: It queries from the database. Okey

File: rotkehlchen/tests/api/test_current_assets_price.py (hunk 4 above)
Comment (yabirgb) [reply rotkehlchen/tests/api/test_current_assets_price.py:540]: I have now the correct indentation in the editor but might have edited it myself

File: rotkehlchen/chain/ethereum/modules/nft/nfts.py (hunk 3 above)
Comment (yabirgb) [reply rotkehlchen/chain/ethereum/modules/nft/nfts.py]: I have changed it to make the field nullable and then the logic makes sense again. Also added the info to the docstring

File: rotkehlchen/db/filtering.py [hunk 5]
```diff
@@ -852,13 +852,20 @@ def make(
 
 @dataclass(init=True, repr=True, eq=True, order=False, unsafe_hash=False, frozen=False)
 class DBNullFilter(DBFilter):
-    columns: list[str]
+    null_columns: Optional[list[str]] = None
```
Comment (LefterisJP) [inline rotkehlchen/db/filtering.py:855]: Why this change? Isn't it more expressive to just have a different filter for the not null? Do you at any point combine those? So have something with both null_columns and not_null_columns?

Comment (yabirgb) [reply rotkehlchen/db/filtering.py:855]: No, I don't combine them in the current filters. Will then split then it in two

**Review by LefterisJP — APPROVED** [review]
Perfect. Nice thing with the verb. LGTM.

File: rotkehlchen/db/filtering.py (hunk 5 above)
Comment (LefterisJP) [reply rotkehlchen/db/filtering.py]: makes sense. Thanks!
